lib/auth/auth.go

```diff
@@ -2152,6 +2152,11 @@ func (a *Server) GenerateHostCerts(ctx context.Context, req *proto.HostCertsRequ
 		NotAfter:  a.clock.Now().UTC().Add(defaults.CATTL),
 		DNSNames:  append([]string{}, req.AdditionalPrincipals...),
 	}
+
+	// DefaultParticipantForRole requests need to specify a DNS name, which must be present in the certificate's DNS Names.
+	// The target DNS is not always known in advance so we add a default one to all certificates.
+	certRequest.DNSNames = append(certRequest.DNSNames, DefaultParticipantForRole(req.Role)...)
+
 	// API requests need to specify a DNS name, which must be present in the certificate's DNS Names.
 	// The target DNS is not always known in advance so we add a default one to all certificates.
 	if (types.SystemRoles{req.Role}).IncludeAny(types.RoleAuth, types.RoleAdmin, types.RoleProxy, types.RoleKube, types.RoleApp) {
```

Comment:
Maybe we should remove this if the new function now takes care of it

Reply:
I think that this is not fully safe. 

Right now the APIDomain DNS for Kube and App role is added on the server side. 
If we remote this we will break logic for old version of the app/kube agents.

Reply:
No I meant the second append at lines 2162-2164

Reply:
Also, should we move this before the SSH certificate generation?

Reply:
The `certRequest` is only used for `tlsAuthority.GenerateCertificate`   call so don't don't see any benefits of moving this before SSH cert generation.